pkg/cloudprovider/vsphereparavirtual/cloud.go

```diff
@@ -167,7 +167,8 @@ func (cp *VSphereParavirtual) Routes() (cloudprovider.Routes, bool) {
 
 // ProviderName returns the cloud provider ID.
 func (cp *VSphereParavirtual) ProviderName() string {
-	return ProviderName
+	// used for constructing ProviderID.
+	return "vsphere"
```

Comment:
will this cause conflicts somewhere else because it's different from the name for registration?

Reply:
This `ProviderName()` func is used in these place for now
https://github.com/kubernetes/cloud-provider/blob/2c911671bd3eeff09ec2e093a5544ed56dd7f06f/cloud.go#L114
https://github.com/kubernetes/cloud-provider/blob/67ac9f4d3f7b3bec927df636c4573e1159eaac30/controllers/node/node_controller.go#L414
https://github.com/kubernetes/cloud-provider/blob/dad6989f990bb1a974d22e9765bb3b7fb83250c9/controllers/service/controller.go#L413


`ProviderName` variable is only used when register 
https://github.com/kubernetes/cloud-provider/blob/03b2e3fe9feb25c0620c23fbc3bf43925e377a9c/plugins.go#L55
https://github.com/kubernetes/cloud-provider/blob/03b2e3fe9feb25c0620c23fbc3bf43925e377a9c/plugins.go#L79
It's only stored in this map and is not used anywhere else

So there shouldn't be conflicts